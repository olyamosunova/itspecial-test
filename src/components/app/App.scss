body {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  margin: 0;
  padding: 50px 0;

  font-family: Avenir, Helvetica, Arial, sans-serif;

  button,
  input,
  textarea {
    font-family: inherit;

    resize: none;
  }
}

#app {
  flex-grow: 1;
  display: flex;
  flex-direction: column;

  & > section {
    flex-grow: 1;
    display: flex;
    flex-direction: column;
  }
}

.visually-hidden {
  position: absolute !important;
  width: 1px !important;
  height: 1px !important;
  padding: 0 !important;
  margin: -1px !important; // Fix for https://github.com/twbs/bootstrap/issues/25686
  overflow: hidden !important;
  clip: rect(0, 0, 0, 0) !important;
  white-space: nowrap !important;
  border: 0 !important;
}
