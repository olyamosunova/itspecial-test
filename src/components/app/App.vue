<template>
  <div id="app">
    <Main />
  </div>
</template>

<script>
  import "./App.scss";
  import 'bootstrap/dist/css/bootstrap.css';
  import 'bootstrap-vue/dist/bootstrap-vue.css';
  import Main from "../main/Main";

export default {
  name: 'App',
  components: {
    Main
  }
}
</script>
