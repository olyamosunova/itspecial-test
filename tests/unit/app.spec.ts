import { mount, createLocalVue } from "@vue/test-utils";
import App from "@/components/app/App.vue";
import Vuex from "vuex";

const localVue = createLocalVue();
localVue.use(Vuex);

describe('unit tests for App component', () => {
    let store: any;
    let actions;

    beforeEach(() => {
        actions = {
            getData: jest.fn(),
        };

       store = new Vuex.Store({ actions });
    });

    it('should match the snapshot', () => {
        const wrapper = mount(App, { store, localVue });
        expect(wrapper.html()).toMatchSnapshot();
    });
});
