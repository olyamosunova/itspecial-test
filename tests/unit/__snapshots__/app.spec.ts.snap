// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit tests for App component should match the snapshot 1`] = `
<div id="app">
  <section class="container">
    <form>
      <div class="input-group mb-3"><input type="text" placeholder="Enter text" aria-label="Enter text" required="required" class="form-control"> <button type="submit" disabled="disabled" class="btn btn-primary">Find</button></div>
    </form>
    <div class="loader d-flex justify-content-center">
      <div role="status" class="spinner-border"><span class="visually-hidden">Loading...</span></div>
    </div>
    <!---->
  </section>
</div>
`;
